### ISSUE TITLE
Fail to reproduce the paper results

### ISSUE THREAD

**Issue by Nusselder9** [NM-sparsity/NM-sparsity#4]
I really appreciate the authors for doing such an inspiring job and sharing the code.

I run this code on 4-2 sparsity Res50 with this code and get accuracy as **75.8%**, which fails to reproduce the paper result (**77.0%**) with the same settings. Can you give me some advices. Thanks!

My pytorch version is
```
>>> torch.__version__
'1.6.0+cu101'
```

**Comment by Nusselder9** [comment]
Here is the detailed results, I upload it in my fork because of the maximum character limit of issues.<br>
ResNet18 **70.46%** vs. paper's **71.2%**. [res18](https://github.com/Nusselder9/NM-sparsity/blob/main/res18.md)<br>
ResNet50 **75.8%** vs. paper's **77.0%**. [res50](https://github.com/Nusselder9/NM-sparsity/blob/main/res50.md)

**Comment by ilil96** [comment]
I am also having a similar issue with reproducing the imagenet result...

**Comment by aojunzz** [comment]
@Nusselder9 @ilil96 It seems strange,  I will check the code and upload the model zoo in several days. I run the ResNet18 model using Sparse_NHWC  yesterday, @Nusselder9  the log https://github.com/NM-sparsity/NM-sparsity/blob/main/res18_2:4.log

**Comment by aojunzz** [comment]
@Nusselder9 @ilil96 Hi, I clone the code and run the res50 model with ImageNet, the log (https://drive.google.com/file/d/1QslUFmU8Eam8j-r1roN11hdwLZtNGrJs/view?usp=sharing), the model (https://drive.google.com/file/d/1U3P4xNbCh4VDTKe24_RFscTfFd42tLAT/view?usp=sharing).

@sunlex0717 has run the pre-training model with his ImageNet validation datasets, the result can achieve 76.9%.

**Comment by Nusselder9** [comment]
Thanks for your kindly reply and your inspiring work again. I will check my datasets. And can you give me some advice about PyTorch version?

**Comment by aojunzz** [comment]
@Nusselder9 PyTorch version is 1.3.

**Comment by lockshaw** [comment]
@Nusselder9 @ilil96 Did either of you eventually figure out the source of the lower accuracy? I'm running into the same issue (getting 75.8% accuracy on resnet50 instead of 77.3%, also running a new version of pytorch)

**Comment by aojunzz** [comment]
@lockshaw Hi, I don't notice your question, sorry for response too late, can you reproduce the accuracy on resnet50 now?

**Comment by lockshaw** [comment]
@NM-sparsity Thanks for the response! No, I unfortunately wasn't able to reproduce the accuracy, or to track down exactly what was causing the difference. The results I was getting _did_ appear qualitatively similar (in terms of gap between dense and sparse results) to those reported

**Comment by CYang0515** [comment]
Is the [model](https://drive.google.com/file/d/1U3P4xNbCh4VDTKe24_RFscTfFd42tLAT/view?usp=sharing) resnet50_dense model? We test the model on imagenet validate set and only get 75.242% top-1 accuracy(not 76.9%) . And We also test torchvision pretrained model and get same results as [torchvision](https://pytorch.org/vision/stable/models.html). Is there difference between dataset or preprocess from torchvision?

**Comment by CYang0515** [comment]
> @Nusselder9 @ilil96 It seems strange, I will check the code and upload the model zoo in several days. I run the ResNet18 model using Sparse_NHWC yesterday, @Nusselder9 the log https://github.com/NM-sparsity/NM-sparsity/blob/main/res18_2:4.log

@NM-sparsity Hi, I run the resnet18 using sparse_NHWC and get 70.944% top1 accuracy (similar to 71.152% provided by author) in train.log, but when evaluating the model after training, I only get  70.366% top1 accuracy. Please is it normal? And the problem is caused by distributed training?
I also test the provided Sparse_NHWC resnet50 [model](https://drive.google.com/file/d/15uBFc7-7PEpLQg6ZCUnXXVMDOSlCddsx/view) and get 76.91% top-1 accuracy, not 77.1% top1 accuracy.

**Comment by aojunzz** [comment]
@CYang0515 the gap between training-evaluation and evaluation-only shouldn't be caused by distribution training.

**Comment by CYang0515** [comment]
@NM-sparsity The 77.1% top1 accuracy of provided Sparse_NHWC resnet50 [model](https://drive.google.com/file/d/15uBFc7-7PEpLQg6ZCUnXXVMDOSlCddsx/view) is evaluated on  training-evaluation or evaluation-only? I evaluate the torchvision model and get the accurate accuracy, so there are any difference between two preprocess?

**Comment by JingyangXiang** [comment]
> @NM-sparsity The 77.1% top1 accuracy of provided Sparse_NHWC resnet50 [model](https://drive.google.com/file/d/15uBFc7-7PEpLQg6ZCUnXXVMDOSlCddsx/view) is evaluated on training-evaluation or evaluation-only? I evaluate the torchvision model and get the accurate accuracy, so there are any difference between two preprocess?

I guess it causes by the BatchNorm2d layer, the code use different mean and var on different GPUs if SyncBN is not applied

**Comment by hopef** [comment]
Hi,
I'm using the latest code and can't reproduce the paper's results (77.1%).  
```bash
 * Prec@1 76.480 Prec@5 92.880
```
Here is my training log: [baseline-20231024_080813.log](https://github.com/aojunzz/NM-sparsity/files/13199948/baseline-20231024_080813.log)

Has this issue not been resolved?
Does it mean that this method is unstable?

**Comment by hopef** [comment]
For ResNet50: 
I can easily reproduce the accuracy for dense (77.1%) using your code.
But I never reproduce the accuracy for sparsity4:2(77.0%).

1. In your paper, the accuracy of dense is 77.3%, why the reproduction is 77.1% only?
2. In your paper, the accuracy of sparsity4:2 is 77.0%, why the reproduction is 76.480% only?

Why is there such a big gap?
Does it mean that this method is very unstable?

**Comment by hopef** [comment]
Your paper lacks credibility!
